lms/static/js/dashboard/views/course_list_view.js

```diff
@@ -0,0 +1,39 @@
+;(function (define) {
+    'use strict';
+
+    define(['jquery', 'underscore', 'common/js/components/views/list', 'text!templates/dashboard/courses.underscore'],
+        function ($, _, ListView, coursesTemplate) {
+
+            var CourseListView = ListView.extend({
+
+                initialize: function (options) {
+                    this.itemViewClass = options.itemViewClass || this.itemViewClass;
+                    this.template = _.template(coursesTemplate);
+                    this.parent = options.parent;
+                    this.settingsModel = options.settingsModel;
+                    this.itemViews = [];
+                },
+
+                render: function () {
+                    this.$el.html(this.template({courses: this.collection, settings: this.settingsModel}));
+                    this.collection.each(this.createItemView, this);
+
+                    return this;
+                },
+
+                createItemView: function (course) {
+                    var itemView = new this.itemViewClass({
+                        model: course,
+                        settingsModel: this.settingsModel,
+                        tabbedView: this.parent
+                    });
+
+                    this.$('.listing-courses').append(itemView.render().el);
+                    this.itemViews.push(itemView);
+                }
```

Comment:
Instead of appending each view one at a time to the DOM you could add them to DOM element in memory and then push that into the actual DOM once all have been appended (to follow your current approach).